### PR TITLE
EUI-7160 Case file view hand icon for the folder clicking

### PR DESCRIPTION
**Pull request by johnbenjamin-hmcts**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/EUI-7066
https://tools.hmcts.net/jira/browse/EUI-7160
https://tools.hmcts.net/jira/browse/EUI-5458

### Change description ###
Case file view: Search document exceptions
Case file view: Hand icon for the folder clicking
Case file view: Document view & folder action

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[X] No
```

### CODE REVIEW

File: projects/ccd-case-ui-toolkit/src/lib/shared/domain/case-file-view/document-tree-node.model.ts [hunk 1]
```diff
@@ -4,3 +4,8 @@ export interface DocumentTreeNode {
   count?: number;
```
Comment (MateiHMCTS) [inline projects/ccd-case-ui-toolkit/src/lib/shared/domain/case-file-view/document-tree-node.model.ts:4]: Maybe make type as DocumentTreeNodeType as well?

File: projects/ccd-case-ui-toolkit/src/lib/shared/components/palette/case-file-view/components/case-file-view-folder/case-file-view-folder.component.ts [hunk 2]
```diff
@@ -16,13 +19,20 @@ import {
...
+  private static readonly DOCUMENT_SEARCH_FORM_CONTROL_NAME = 'documentSearchFormControl';
+  private static readonly MINIMUM_SEARCH_CHARACTERS = 3;
 
   @Input() public categoriesAndDocuments: Observable<CategoriesAndDocuments>;
 
   public nestedTreeControl: NestedTreeControl<DocumentTreeNode>;
   public nestedDataSource: DocumentTreeNode[];
   public categories: CaseFileViewCategory[] = [];
   public categoriesAndDocumentsSubscription: Subscription;
+  public documentFilterFormGroup: FormGroup;
```
Comment (MateiHMCTS) [inline projects/ccd-case-ui-toolkit/src/lib/shared/components/palette/case-file-view/components/case-file-view-folder/case-file-view-folder.component.ts:31]: You can probably get away without using a FormGroup. I can only see a single FormControl, therefore a FormGroup isn't needed.

**Review by MateiHMCTS — APPROVED** [review]

File: projects/ccd-case-ui-toolkit/src/lib/shared/components/case-viewer/case-full-access-view/case-full-access-view.component.html [hunk 3]
```diff
@@ -59,14 +59,14 @@ <h2 class="heading-h2 error-summary-heading" id="edit-case-event_error-summary-h
                       <th id="case-viewer-field-label" *ngIf="!isFieldToHaveNoLabel(field)">
                         <div class="case-viewer-label text-16">{{field.label}}</div>
                       </th>
-                      <th>
+                      <td>
```
Comment (Daniel-Lam) [inline projects/ccd-case-ui-toolkit/src/lib/shared/components/case-viewer/case-full-access-view/case-full-access-view.component.html:62]: @johnbenjamin-hmcts - the `<td>` is back again...?

**Review by Daniel-Lam — CHANGES_REQUESTED** [review]
@johnbenjamin-hmcts - just a minor comment from me... it's that `<td>` tag again!

File: projects/ccd-case-ui-toolkit/src/lib/shared/components/case-viewer/case-full-access-view/case-full-access-view.component.html (hunk 3 above)
Comment (johnbenjamin-hmcts) [reply projects/ccd-case-ui-toolkit/src/lib/shared/components/case-viewer/case-full-access-view/case-full-access-view.component.html:62]: yes @Daniel-Lam, as we discussed earlier it should be `td` and not `th` as per the ticket https://tools.hmcts.net/jira/browse/EUI-6551 you mentioned during our chat on 11th November in team-x channel.

File: projects/ccd-case-ui-toolkit/src/lib/shared/domain/case-file-view/document-tree-node.model.ts (hunk 1 above)
Comment (johnbenjamin-hmcts) [reply projects/ccd-case-ui-toolkit/src/lib/shared/domain/case-file-view/document-tree-node.model.ts:4]: Thank you @MateiHMCTS, I have now made the changes to the code.

**Review by MateiHMCTS — APPROVED** [review]

**Pull request merged**